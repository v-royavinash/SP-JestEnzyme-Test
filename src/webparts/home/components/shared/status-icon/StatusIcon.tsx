import React from "react";

export interface IStausIconProps {
    iconName: string;
}

const StausIcon = ({ iconName }: IStausIconProps) => {
    const inprogressIcon: string = require("../../../assets/images/inprogress.png");
    const completedIcon: string = require("../../../assets/images//completed.png");
    const notstartedIcon: string = require("../../../assets/images//notstarted.png");

    return (
        <React.Fragment>
            {
                {
                    inprogress: (
                        <img src={inprogressIcon} />
                    ),
                    completed: (
                        <img src={completedIcon} />
                    ),
                    notstarted: (
                        <img src={notstartedIcon} />
                    ),
                }[iconName]
            }
        </React.Fragment>
    );
};

export default StausIcon;
