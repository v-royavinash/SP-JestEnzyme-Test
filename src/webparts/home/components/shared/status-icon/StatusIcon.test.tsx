/// <reference types="jest" />

import React from 'react';
import { mount, ReactWrapper } from 'enzyme';

import { IStausIconProps } from './StatusIcon';
import StatusIcon from './StatusIcon';

describe('Shared Component : StatusIcon Unit Test', () => {

    let iconName: string = "inprogress";
    let reactComponent: ReactWrapper<IStausIconProps, {}>;

    beforeEach(() => {
        reactComponent = mount(React.createElement(
            StatusIcon, { iconName: iconName }
        ));
    });

    afterEach(() => reactComponent.unmount());

    test('Component should render a image tag', () => {
        expect(reactComponent.find('img')).not.toBeNull();
    });

    test('Component should return only one a Image tag', () => {
        expect(reactComponent.find('img').length).toEqual(1);
    });

    test('Component should match the UI snapshot', () => {
        expect(reactComponent.html).toMatchSnapshot();
    });
});