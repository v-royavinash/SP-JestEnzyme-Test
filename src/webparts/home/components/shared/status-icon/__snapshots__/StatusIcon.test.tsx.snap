// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Component : StatusIcon Unit Test Component should match the UI snapshot 1`] = `[Function]`;
